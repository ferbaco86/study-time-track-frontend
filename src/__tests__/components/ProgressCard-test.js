import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/store';
import ProgressCard from '../../components/ProgressCard';

describe('Rendering component', () => {
  it('creates a ProgressCard component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <ProgressCard
              date="01/01/01"
              name="test"
              time={10}
            />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('10');
  });
});

describe('ProgressCard', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <ProgressCard
            date="01/01/01"
            name="test"
            time={10}
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
