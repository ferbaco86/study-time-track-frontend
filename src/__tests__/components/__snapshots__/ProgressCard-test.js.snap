// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressCard renders correctly 1`] = `
<div
  className="sc-bdfBwQ cOFkCz"
>
  <div
    className="sc-jSgupP jkFoLt"
  >
    <h2
      className="sc-hKgILt djViUa"
    >
      01/01/01
    </h2>
    <h4
      className="sc-gsTCUz ctLQUY"
    >
      test
    </h4>
  </div>
  <div
    className="sc-eCssSg fpbVIu"
  >
    <h2
      className="sc-dlfnbm jJeQ"
    >
      10
    </h2>
    <span>
      hr(s)
    </span>
  </div>
</div>
`;
